lib/cartopy/mpl/patch.py

```diff
@@ -249,3 +265,183 @@ def not_zero_poly(geom):
     result = list(filter(not_zero_poly, geom_collection))
 
     return result + other_result_geoms
+
+
+def path_to_shapely(path, force_ccw=False):
+    """
+    Create a Shapely geometric objects from a :class:`matplotlib.path.Path`.
+
+    Parameters
+    ----------
+    path
+        A :class:`matplotlib.path.Path` instance.
+
+    Other Parameters
+    ----------------
+    force_ccw
+        Boolean flag determining whether the path can be inverted to enforce
+        ccw. Defaults to False.
+
+    Returns
+    -------
+    One of the following Shapely objects:
+
+        :class:`shapely.geometry.polygon.Polygon`,
+        :class:`shapely.geometry.linestring.LineString`
+        :class:`shapely.geometry.point.Point`,
+        :class:`shapely.geometry.multipolygon.MultiPolygon`
+        :class:`shapely.geometry.multilinestring.MultiLineString`
+        :class:`shapely.geometry.multipoint.MultiPoint`
+        :class:`shapely.geometry.collection.GeometryCollection`.
+
+    """
+    # Convert path into numpy array of vertices (and associated codes)
```

Comment:
Are we able to leverage any of the recent contour logic / path handling you were updating in Matplotlib?

https://github.com/matplotlib/matplotlib/blob/25a05f2399d6b63bbaad5a9380a8a89cec821c50/lib/matplotlib/contour.py#L795

There are a bunch of path handling routines for stitching the previous individual paths into a compound path with the recent refactor that might give some inspiration here as well.

Reply:
ContourSet's `allsegs` and `allkinds` use the private `Path._iter_connected_components`.  We could raise a feature request to have that made public.  However, I made a quick test swapping out our path splitting logic here for `_iter_connected_components` and got some spectacular image test failures.  I didn't dig too deep, but it seems some of the `_iter_connected_components` paths have an extra `STOP` on the end which our own splitting logic doesn't produce.

Or have I misunderstood what you meant?